src/controllers/Order.controller.ts

```diff
@@ -0,0 +1,144 @@
+      where: {
+        userId,
+      },
+    });
+    const items = cartItem.map((item) => item.dataValues.productId);
+    if (items.length === 0) {
+      return res
+        .status(400)
+        .json({ message: "No product available To make order" });
+    }
+    const order = await Order.create({
+      userId: userId,
+      items: JSON.stringify(items),
+    });
+    res.status(201).json({
+      message: "Order created",
+      orderData: order,
+    });
+  } catch (error) {
+    if (error instanceof Error) {
+      res.status(500).json({
+        status: 500,
+        success: false,
+        message: "Error creating order",
+        error: error.message,
+      });
+    }
+  }
+};
+
+export const ViewOrders = async (req: Request, res: Response) => {
+  try {
+    const userId = req.user.id;
+    const orders = await Order.findAll({
+      where: {
+        userId,
+      },
+    });
+
+    res.status(200).send({
+      status: 200,
+      success: true,
+      data: orders,
+    });
+  } catch (error) {
+    if (error instanceof Error) {
+      res.status(500).json({
+        status: 500,
+        success: false,
+        message: "Error  while getting order",
+        error: error.message,
+      });
+    }
+  }
+};
+
+// clear the cart
+export const clearOrders = async (req: Request, res: Response) => {
+  try {
+    const userId = req.user.id;
+
+    const clearOrder = await Order.destroy({
+      where: {
+        userId,
+      },
+    });
+
+    res.status(201).send({
+      status: 201,
+      message: "All Order cancered successfully!",
+      data: clearOrder,
+    });
+  } catch (error) {
+    if (error instanceof Error) {
+      res.status(500).json({
+        status: 500,
+        success: false,
+        message: "Error while cancering order",
+        error: error.message,
+      });
+    }
+  }
+};
+
+// GET request to retrieve order status information
+export const getOrderStatus = async (req: Request, res: Response) => {
+  try {
+    const userId = req.user.id;
+    const orderId = req.params.orderId;
+
+    const orders = await Order.findAll({
+      where: {
+        userId,
+      },
+    });
+
+    const order = orders.find((ava) => ava.dataValues.id === orderId);
+    if (!order) {
+      return res.status(404).json({ message: "Order not found" });
+    }
+    const orderStatus = {
+      currentStatus: order.dataValues.status,
+      expectedDeliveryDate: order.dataValues.expectedDeliveryDate,
+    };
+
+    return res.status(200).json(orderStatus);
+  } catch (err) {
+    return res.status(500).json({ message: "Internal server error" });
+  }
+};
+
+// PUT request to update order status information
+export const updatedOrderStatus = async (req: Request, res: Response) => {
+  try {
+    const orderId = req.params.orderId;
+    const order = await Order.findOne({
+      where: {
+        id: orderId,
+      },
+    });
+    if (!order) {
+      return res.status(404).json({ message: "Order not found" });
```

Comment:
These response messages are not the ones we using to be honest. This is way off track, wouldn't you agree❓